### ISSUE TITLE
The ShouldConstrainToRootBounds property of the Popup doesn't work

### ISSUE THREAD

**Issue by alexdi220** [microsoft/microsoft-ui-xaml#5958]
**Describe the bug**
How can I show a popup out of the window's bounds? The `ShouldConstrainToRootBounds` looks pretty good ([doc](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.primitives.popup.shouldconstraintorootbounds?view=winui-3.0#remarks)) but doesn't work and the `IsConstrainedToRootBounds` property always returns `true`

**Steps to reproduce the bug**
```xaml
    <StackPanel Orientation="Horizontal" HorizontalAlignment="Center" VerticalAlignment="Center">
        <Button Content="Click" Click="Button_Click"/>
        <Popup x:Name="popup" ShouldConstrainToRootBounds="False">
            <Border Width="100" Height="500" Background="Red" BorderThickness="3" BorderBrush="Black"/>
        </Popup>
    </StackPanel>
```
```csharp
    public sealed partial class MainWindow : Window {
        public MainWindow() {
            this.InitializeComponent();
        }
        void Button_Click(object sender, RoutedEventArgs e) {
            popup.IsOpen = !popup.IsOpen;
        }
    }
```

**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/47348109/134667281-19e55099-46fe-4623-8560-4ff1953b834c.png)

![image](https://user-images.githubusercontent.com/47348109/134667137-beed3db6-ff19-4bf3-9950-e7bf35c08d13.png)

**Version Info**

NuGet package version: 

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  | Yes |

| Windows version                    | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2021 Update (19043)            |    |
| October 2020 Update (19042)        |    |
| May 2020 Update (19041)            |  Yes  |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            |  Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

**Comment by codendone** [comment]
`ShouldConstrainToRootBounds="False"` is not yet supported in WinUI 3.

**Comment by alexdi220** [comment]
Thanks, @codendone . Could you add a note\remark in the documentation about that?

**Comment by krschau** [comment]
@gabbybilka Can we update the doc above to reflect that False is not yet supported? Thanks!

**Comment by gabbybilka** [comment]
@krschau, yes!

**Comment by Samuel12321** [comment]
@krschau why was this closed when it hasn't been fixed yet? 
`ShouldConstrainToRootBounds="False"` Still doesn't work in winui 3.

**Comment by krschau** [comment]
@Samuel12321 We should be enabling this soon, although it won't be quite soon enough for the WinAppSDK 1.0 release. As I mentioned when I closed it, I think that since it's now appropriately documented, this particular issue can be closed. This doesn't mean that we aren't working on enabling the feature. 

I see pros and cons in closing or leaving issues open issues in situations like this, and we'll adjust if we see that the way we're doing it isn't helpful as we go forward.

**Comment by asklar** [comment]
Tracked internally by https://task.ms/36484033

**Comment by Samuel12321** [comment]
Will a fix for this be included for 1.1?

**Comment by codendone** [comment]
@Samuel12321 Additional feature work was necessary to properly support ShouldConstrainToRootBounds=False. That work is in progress, but it looks like it won't be complete in time for 1.1.

**Comment by Samuel12321** [comment]
@codendone is there any update as to when this issue will be fixed. Its currently blocking my team from using WINUI 3.

**Comment by codendone** [comment]
@Samuel12321 Nothing new here from my previous comment: This unfortunately won't be enabled in 1.1. I'm hoping it will be in 1.2, but can't say for sure at this time.

**Comment by Samuel12321** [comment]
@codendone is there any chance this will make it to one of the 1.1 servicing releases?
Sorry to bug, but this is a hugely problematic issue for my team and a few other projects I'm aware of.

**Comment by krschau** [comment]
@Samuel12321 I expect this wouldn't be appropriate for 1.1 servicing, because behind the scenes this is a pretty substantial change. We understand it's a pretty huge pain point, though, and are actively working on it.

**Comment by HannesHold** [comment]
Looking forward have such a functionality available for WinUI3 apps :)

**Comment by Samuel12321** [comment]
@krschau Will this be part of 1.2?

**Comment by Samuel12321** [comment]
@codendone @krschau will this be part of 1.2?

**Comment by codendone** [comment]
@Samuel12321 Unfortunately this will not be in 1.2. Most of the implementation is complete, but there are a few critical issues which still need to be addressed.

**Comment by AlexanderBlackman** [comment]
I really hope this'll come sooner rather than later.  It's probably the only feature I'd even use an experimental build for.

**Comment by AestheticalZ** [comment]
Been almost 2 years since this issue was opened and it still hasn't been solved, even though apps like Files would benefit GREATLY from it. Is there any ETA or will there continue to be silence until someone else comes and complains about it?

**Comment by sylveon** [comment]
I also agree. I am currently stuck on XAML Islands primarily because WinUI 3 doesn't implement this.

**Comment by yaira2** [comment]
@gabbybilka are there any updates on this issue? It's one of the main issues we're having since migrating to WinAppSdk.

**Comment by TheXamlGuy** [comment]
Will this ever be fixed?

**Comment by HannesHold** [comment]
No Assignees, No Projects and No Milestones yet. - Quite a good time period passed and it's getting frustrating and it feels like writing into a void. - Is there someone out there who actually can help out about this?

**Comment by codendone** [comment]
This continues to be a work in progress. It still isn't ready to be part of the upcoming 1.3 release, but the goal is to complete it for 1.4.

**Comment by yaira2** [comment]
@codendone thank you for the transparency on the timeline. We've been waiting for a long time for this to be fixed and 1.4 is too far away, it's not the only issue we've had but most of the time we're able to work around them. This issue, however, doesn't have any known workarounds and it's one of the top frustrations for our user base.

I hope you can take this feedback into consideration and see if there is any way to increase the priority for getting this into a servicing release so that we don't have to wait for 1.4.

**Comment by andrezzzini** [comment]
this feature is really missed.

**Comment by zhuxb711** [comment]
1.3 fixed nothing important, most bugs reported to WinUI team is being ignored.

It's too inefficient and I hope .Net team could hand over the repo one day as they are very efficient and never delay their due date. For WinUI team: Oh it's too hard and let's arrange it to next release (Repeat it); For .Net team: Oh it's too hard let's gathering around to solve it!

Sorry for the complaints and that's what I thought to WinUI team with very disappointment.

**Comment by sylveon** [comment]
WinUI is a C++ library not a .NET one.

**Comment by AlexanderBlackman** [comment]
> This issue, however, doesn't have any known workarounds and it's one of the top frustrations for our user base.
Exactly.
The closest hypothetical workaround I've thought of is creating a new temporary window with just the menu options, but even that would be more trouble than it is worth. This has been an issue for years.

**Comment by Gaoyifei1011** [comment]
> **Describe the bug** How can I show a popup out of the window's bounds? The `ShouldConstrainToRootBounds` looks pretty good ([doc](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.primitives.popup.shouldconstraintorootbounds?view=winui-3.0#remarks)) but doesn't work and the `IsConstrainedToRootBounds` property always returns `true`
> 
> **Steps to reproduce the bug**
> 
> ```
>     <StackPanel Orientation="Horizontal" HorizontalAlignment="Center" VerticalAlignment="Center">
>         <Button Content="Click" Click="Button_Click"/>
>         <Popup x:Name="popup" ShouldConstrainToRootBounds="False">
>             <Border Width="100" Height="500" Background="Red" BorderThickness="3" BorderBrush="Black"/>
>         </Popup>
>     </StackPanel>
> ```
> 
> ```cs
>     public sealed partial class MainWindow : Window {
>         public MainWindow() {
>             this.InitializeComponent();
>         }
>         void Button_Click(object sender, RoutedEventArgs e) {
>             popup.IsOpen = !popup.IsOpen;
>         }
>     }
> ```
> 
> **Expected behavior**
> 
> **Screenshots** ![image](https://user-images.githubusercontent.com/47348109/134667281-19e55099-46fe-4623-8560-4ff1953b834c.png)
> 
> ![image](https://user-images.githubusercontent.com/47348109/134667137-beed3db6-ff19-4bf3-9950-e7bf35c08d13.png)
> 
> **Version Info**
> 
> NuGet package version:
> 
> Windows app type:
> 
> UWP	Win32
> Yes
> Windows version	Saw the problem?
> Insider Build (xxxxx)	
> May 2021 Update (19043)	
> October 2020 Update (19042)	
> May 2020 Update (19041)	Yes
> November 2019 Update (18363)	
> May 2019 Update (18362)	
> October 2018 Update (17763)	
> April 2018 Update (17134)	
> Fall Creators Update (16299)	
> Creators Update (15063)	
> Device form factor	Saw the problem?
> Desktop	Yes
> Xbox	
> Surface Hub	
> IoT	
> **Additional context**

in windows app sdk 1.4-experimental 1 it is successfully work

**Comment by Samuel12321** [comment]
Can we get some confirmation on whether this has been addressed in 1.4 experimental 1 from the wasdk dev team.

**Comment by Pietro228** [comment]
Now it does work!

**Comment by ramjke** [comment]
> In 1.4-experimental 1 it does work!

Confirmed. But there's a new problem. Now backdrop is also disabled when ShouldConstrainToRootBounds = False.

https://github.com/microsoft/microsoft-ui-xaml/issues/8657

**Comment by khoabui1412** [comment]
MenuFlyoutItem doesn't have ShouldConstrainToRootBounds property, how to deal with that?

**Comment by sylveon** [comment]
The property is on MenuFlyout itself

**Comment by khoabui1412** [comment]
@sylveon, I tested but it doesn't work for MenuFlyoutItem though set ShouldConstrainToRootBounds=false in MenuFlyout

**Comment by khoabui1412** [comment]
I clean the bin/obj, close vs and re-open, it's ok now.
sorry for make confused

**Comment by codendone** [comment]
Closing as fixed in 1.4.
